client/containers/EventsListPanelContainer/index.js

```diff
@@ -39,14 +40,30 @@ class EventsListPanel extends React.Component {
         this.setState({ searchInputValue: event.target.value })
     }
 
+    toggleAdvancedSearch() {
```

Comment:
Nothing wrong with this. I just share how I would have done this.
1. Add a `toggleAdvSearch` in `mapStateToProps`
2. Implement this toggle in it, without touching the component state but by using the value directly from the state
3. Call this `toggleAdvSearch` property on the icon `onClick`

This is not better, it's just another way that prevents to create a component state.